clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/spring/client/configuration/CredentialsProviderConfiguration.java

```diff
@@ -39,6 +40,37 @@ public class CredentialsProviderConfiguration {
   @ConditionalOnMissingBean
   public CredentialsProvider camundaClientCredentialsProvider(
       final CamundaClientProperties camundaClientProperties) {
+    final var clientMode = camundaClientProperties.getMode();
+
+    return clientMode == null
+        ? new NoopCredentialsProvider()
+        : switch (clientMode) {
+          case basic -> buildBasicAuthCredentialsProvider(camundaClientProperties);
+          case saas, selfManaged -> buildOAuthCredentialsProvider(camundaClientProperties);
+        };
+  }
+
+  private CredentialsProvider buildBasicAuthCredentialsProvider(
+      final CamundaClientProperties camundaClientProperties) {
+    final var username = camundaClientProperties.getAuth().getUsername();
+    final var password = camundaClientProperties.getAuth().getPassword();
+
+    final var builder =
+        new BasicAuthCredentialsProviderBuilder()
+            .applyEnvironmentOverrides(false)
+            .username(username)
+            .password(password);
+
+    try {
+      return builder.build();
+    } catch (final Exception e) {
+      LOG.warn("Failed to configure credential provider", e);
+      return new NoopCredentialsProvider();
```

Comment:
As we can now use the noop provider by setting no mode, this could throw to prevent configuration issues.

Reply:
I see.. let's leave it that way.